File: Simplex.Demo/Actions/GetBoolean.cs
using System;
using System.Collections.Generic;
using System.Reactive;
using System.Reactive.Linq;
using System.Text;
using Tmon.Simplex.Actions;
using Tmon.Simplex.Channels;

namespace SimplexDemo.Actions
{
    public class GetBoolean : AbstractAction<bool, bool>
    {
        public Channel ExCh1 { get; }

        public Channel ExCh2 { get; }

        public Channel ExCh3 { get; }

        public Channel ExCh4 { get; }

        public override IObservable<bool> Process(bool param)
        {
            return Observable.Return(param);
        }
    }

    public class GetIntegerString : AbstractAction<int, string>
    {
        public Channel ExCh1 { get; }

        public Channel ExCh2 { get; }

        public Channel ExCh3 { get; }

        public override IObservable<string> Process(int param)
        {
            return Observable.Return(param.ToString());
        }
    }

    public class SendSignal : AbstractAction<Unit>
    {
        public override IObservable<Unit> Process()
        {
            return Observable.Return(Unit.Default);
        }
    }
}

File: simplex.java/src/test/java/kr/co/tmon/simplex/sample/actions/GetBoolean.java
package kr.co.tmon.simplex.sample.actions;

import io.reactivex.Observable;
import kr.co.tmon.simplex.actions.AbstractAction;
import kr.co.tmon.simplex.channels.IChannel;

public class GetBoolean extends AbstractAction<Boolean, Boolean> {

	public IChannel ExCh1;
	
	public IChannel ExCh2;
	
	public IChannel ExCh3;
	
	@Override
	public Observable<Boolean> process(Boolean param) {
		return Observable.just(param);
	}
}
 